virtcontainers/qemu_arm64.go

```diff
@@ -45,6 +48,65 @@ var supportedQemuMachines = []govmmQemu.Machine{
 	},
 }
 
+// Logger returns a logrus logger appropriate for logging qemu-aarch64 messages
+func qemuArmLogger() *logrus.Entry {
+	return virtLog.WithField("subsystem", "qemu-aarch64")
+}
+
+// On ARM platform, we have different GIC interrupt controllers. Different
+// GIC supports different QEMU parameters for virtual GIC and max VCPUs
+var hostGICVersion = getHostGICVersion()
+
+// We will access this file on host to detect host GIC version
+var gicProfile = "/proc/interrupts"
+
+// Detect the host GIC version.
+// Success: return the number of GIC version
+// Failed: return 0
+func getHostGICVersion() (version uint32) {
+	bytes, err := ioutil.ReadFile(gicProfile)
+	if err != nil {
+		qemuArmLogger().WithField("GIC profile", gicProfile).WithError(err).Error("Failed to parse GIC profile")
```

Comment:
Nit: It's an extra line of code, but I'd still be tempted to add `return 0` here to fail fast.